formats/hocon/src/main/kotlin/kotlinx/serialization/hocon/Hocon.kt

```diff
@@ -231,10 +265,25 @@ public class HoconBuilder internal constructor(hocon: Hocon) {
      * Switches naming resolution to config naming convention: hyphen separated.
      */
     public var useConfigNamingConvention: Boolean = hocon.useConfigNamingConvention
+
+    /**
+     * Switches polymorphic serialization to the default array format.
+     * This is an option for legacy HOCON format and should not be generally used.
```

Comment:
Not sure if 'the legacy HOCON format' words are applicable here — HOCON never had polymorphism in the first place, so array-based is a legacy only for JSON. I think we can stick with the words 'legacy polymorphism format'